charts/cadence/templates/server-configmap.yaml

```diff
@@ -12,4 +12,731 @@ data:
+            awsConfig:
+              enabled: {{ .Values.config.persistence.elasticsearch.awsSigning.enabled }}
+              region: {{ .Values.config.persistence.elasticsearch.awsSigning.region | quote }}
+              service: {{ .Values.config.persistence.elasticsearch.awsSigning.service | default "es" | quote }}
+            {{- end }}
+            {{- if .Values.config.persistence.elasticsearch.tls.enabled }}
+            tls:
+              enabled: {{ .Values.config.persistence.elasticsearch.tls.enabled }}
+              {{- if .Values.config.persistence.elasticsearch.tls.caFile }}
+              caFile: {{ .Values.config.persistence.elasticsearch.tls.caFile | quote }}
+              {{- end }}
+              {{- if .Values.config.persistence.elasticsearch.tls.caFiles }}
+              caFiles:
+                {{- range .Values.config.persistence.elasticsearch.tls.caFiles }}
+                - {{ . | quote }}
+                {{- end }}
+              {{- end }}
+              {{- if .Values.config.persistence.elasticsearch.tls.certFile }}
+              certFile: {{ .Values.config.persistence.elasticsearch.tls.certFile | quote }}
+              {{- end }}
+              {{- if .Values.config.persistence.elasticsearch.tls.keyFile }}
+              keyFile: {{ .Values.config.persistence.elasticsearch.tls.keyFile | quote }}
+              {{- end }}
+              {{- if .Values.config.persistence.elasticsearch.tls.serverName }}
+              serverName: {{ .Values.config.persistence.elasticsearch.tls.serverName | quote }}
+              {{- end }}
+              enableHostVerification: {{ .Values.config.persistence.elasticsearch.tls.enableHostVerification | default true }}
+              requireClientAuth: {{ .Values.config.persistence.elasticsearch.tls.requireClientAuth | default false }}
+            {{- end }}
+        {{- end }}
+        {{- end }}
+
+    # Ringpop configuration
+    ringpop:
+      name: {{ .Values.config.ringpop.name | default "cadence" }}
+      bootstrapMode: {{ .Values.config.ringpop.bootstrapMode | default "dns" }}
+      bootstrapHosts:
+        - {{ include "cadence.fullname" . }}-frontend-headless:{{ .Values.frontend.port | default 7933 }}
+        - {{ include "cadence.fullname" . }}-history-headless:{{ .Values.history.port | default 7934 }}
+        - {{ include "cadence.fullname" . }}-matching-headless:{{ .Values.matching.port | default 7935 }}
+        - {{ include "cadence.fullname" . }}-worker-headless:{{ .Values.worker.port | default 7939 }}
+      maxJoinDuration: {{ .Values.config.ringpop.maxJoinDuration | default "30s" }}
+
+    # Cluster configuration
+    clusterGroupMetadata:
+      failoverVersionIncrement: {{ .Values.config.cluster.failoverVersionIncrement | default 10 }}
+      primaryClusterName: {{ .Values.config.cluster.primaryClusterName | default "cluster0" }}
+      currentClusterName: {{ .Values.config.cluster.currentClusterName | default "cluster0" }}
+      clusterGroup:
```

Comment:
By default it makes sense to populate `clusterGroup` with just current cluster info. This would be out of the box single cluster deployment. 
However, when `clusterGroup` is populated in values.yml (not unset) then we should just use that here instead of adding current cluster by default. The config would provide complete list of clusters, not just others.